packages/task/src/browser/task-service.ts

```diff
@@ -997,24 +993,19 @@ export class TaskService implements TaskConfigurationClient {
         terminal.sendText(selectedText);
     }
 
-    async attach(terminalId: number, taskId: number): Promise<void> {
-        // Get the list of all available running tasks.
-        const runningTasks: TaskInfo[] = await this.getRunningTasks();
```

Comment:
If no corresponding `taskInfo` is found, the `attach` function would end up with opening a widget labeled with something like `Task: #23` rather than the label of the task, which is ugly.